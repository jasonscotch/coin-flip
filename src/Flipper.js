import React, { useState } from "react";

import Coin from "./Coin";

const Flipper = (props) => {
    const [isShowing, setIsShowing] = useState(false);
    const [totalFlips, setTotalFlips] = useState(0);
    const [headsCount, setHeadsCount] = useState(0);
    const [tailsCount, setTailsCount] = useState(0);

    const handleFlip = (isHeads) => {
        setTotalFlips(totalFlips + 1);

        if (isHeads) {
            setHeadsCount(headsCount + 1);
        } else {
            setTailsCount(tailsCount + 1);
        }
    };

    const handleClick = () => {
        setIsShowing(true);
        handleFlip(true);
    };

    return (
        <div className="Flipper">
            {isShowing ? (
                <Coin onFlip={handleFlip} />
            ) : (
                <button className="Flipper-btn" onClick={handleClick}>FLIP MEEEE</button>
            )}
            <p>Out of {totalFlips} flips, there have been {headsCount} heads and {tailsCount} tails.</p>
        </div>
    );
    
}

export default Flipper;