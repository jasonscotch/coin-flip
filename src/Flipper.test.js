import { fireEvent, render } from '@testing-library/react';
import Flipper from './Flipper';

beforeEach(function() {
    jest
      .spyOn(Math, "random")
      .mockReturnValueOnce(0.25)
      .mockReturnValueOnce(0.75);
  });
  
 
it('renders without crashing', () => {
    render(<Flipper />);
  });
  
  it('matches snapshot', () => {
    const { asFragment } = render(<Flipper />);
    expect(asFragment()).toMatchSnapshot();
  }); 
  
  it("doesn't show coin when page loads", () => {
    const { queryByTestId } = render(<Flipper />);
  
    expect(queryByTestId("Coin")).toBeNull();
  });

  it("counts when heads", () => {
  const { getByText } = render(<Flipper />);

  const button = getByText("FLIP MEEEE");
  fireEvent.click(button);

  expect(
    getByText("Out of 1 flips, there have been 1 heads and 0 tails.")
  ).toBeInTheDocument();
});

it("counts when tails", () => {
  const { getByText } = render(<Flipper />);

  const button1 = getByText("FLIP MEEEE");
  fireEvent.click(button1);
  
  const button2 = getByText("FLIP MEEEE");
  fireEvent.click(button2);
  fireEvent.click(button2);

  expect(
    getByText("Out of 3 flips, there have been 2 heads and 1 tails.")
  ).toBeInTheDocument();
});

  
  afterEach(function() {
    Math.random.mockRestore();
  });