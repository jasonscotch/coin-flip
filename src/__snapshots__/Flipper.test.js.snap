// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Flipper"
  >
    <button
      class="Flipper-btn"
    >
      FLIP MEEEE
    </button>
    <p>
      Out of 0 flips, there have been 0 heads and 0 tails.
    </p>
  </div>
</DocumentFragment>
`;
